internal/controller/dnsrecord_controller.go

```diff
@@ -73,7 +81,7 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	dnsRecord := previous.DeepCopy()
 
 	if dnsRecord.DeletionTimestamp != nil && !dnsRecord.DeletionTimestamp.IsZero() {
-		if err := r.deleteRecord(ctx, dnsRecord); err != nil {
+		if err = r.deleteRecord(ctx, dnsRecord); err != nil {
 			logger.Error(err, "Failed to delete DNSRecord")
 			return ctrl.Result{}, err
 		}
```

Comment:
do we not also need to requeue for validation here to ensure the record is still gone before removing the finalizer?

Reply:
yes we do and it will be part of #38 (and the WIP PR is here #72 - I'm getting it ready now)

Reply:
ok `deleteRecord` calls applyChanges. But we need to requeue after a delete is done

Reply:
the apply changes will return a timeout and in #72  I'm checking if that is a "validation" timeout I'm requeueing (does this word even exist? :laughing: ) until `applyChanges` won't say that there is nothing to be deleted